pkg/ipamd/ipamd.go

```diff
@@ -759,17 +761,20 @@ func (c *IPAMContext) tryAllocateENI() error {
 // For an ENI, try to fill in missing IPs on an existing ENI with PD disabled
 // try to fill in missing Prefixes on an existing ENI with PD enabled
 func (c *IPAMContext) tryAssignCidrs() (increasedPool bool, err error) {
-	short, _, warmTargetDefined := c.datastoreTargetState()
-	if warmTargetDefined && short == 0 {
+	short, _, warmIPTargetDefined := c.datastoreTargetState()
+	if warmIPTargetDefined && short == 0 {
 		log.Infof("Warm IP target set and short is 0 so not assigning Cidrs (IPs or Prefixes)")
 		return false, nil
 	}
 
-	short, warmTargetDefined = c.datastorePrefixTargetState()
-	if warmTargetDefined && short == 0 {
-		log.Infof("Warm prefix target set and short is 0 so not assigning Cidrs (Prefixes)")
-		return false, nil
+	if !warmIPTargetDefined {
+		shortPrefix, warmTargetDefined := c.datastorePrefixTargetState()
+       		if warmTargetDefined && shortPrefix == 0 {
```

Comment:
Same comment as above

Reply:
TryAssignCidr is called from nodeInit and reconciler(previous commented function), hence we need to add the check in both places. Also the reasoning is same, we need to give priority to WARM_IP_TARGET and then check WARM_PREFIX_TARGET since in the manifest we can have both defined.